go/libraries/doltcore/env/environment.go

```diff
@@ -1016,3 +944,59 @@ func (dEnv *DoltEnv) GetUserHomeDir() (string, error) {
 func (dEnv *DoltEnv) TempTableFilesDir() string {
 	return mustAbs(dEnv, dEnv.GetDoltDir(), tempTablesDir)
 }
+
+// GetGCKeepers returns the hashes of all the objects in the environment provided that should be perserved during GC.
+func GetGCKeepers(ctx context.Context, env *DoltEnv) ([]hash.Hash, error) {
```

Comment:
I think that we don't need this anymore. Working and stage are stored in a `dataset` now, which should protect them from GC. That said, I think we should make that a separate change

Reply:
Added a todo